# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaDevices.cpp
# Commit: 61e34bf4f9ce
# Full Hash: 61e34bf4f9ce76225cdd024f5eb677ef7e9d6761
# Author: Tom Ritter <tom@mozilla.com>
# Date: 2022-03-25 09:38:14
# Description:
#   Bug 1760274: Do not crash in RFP mode if GetWrapper is null r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D141505
# ==============================================================================

diff -r 1eff375a0a81 -r 61e34bf4f9ce dom/media/MediaDevices.cpp
--- a/dom/media/MediaDevices.cpp	Thu Mar 24 22:35:57 2022 +0200
+++ b/dom/media/MediaDevices.cpp	Thu Mar 24 20:18:28 2022 +0000
@@ -692,10 +692,17 @@
 
   if (nsContentUtils::ShouldResistFingerprinting(
           "Guarding the more expensive RFP check with a simple one")) {
-    nsCOMPtr<nsIGlobalObject> global = xpc::NativeGlobal(GetWrapper());
-    nsCOMPtr<nsPIDOMWindowInner> owner = do_QueryInterface(global);
+    nsCOMPtr<nsPIDOMWindowInner> window = GetOwner();
+    auto* wrapper = GetWrapper();
+    if (!window && wrapper) {
+      nsCOMPtr<nsIGlobalObject> global = xpc::NativeGlobal(wrapper);
+      window = do_QueryInterface(global);
+    }
+    if (!window) {
+      return;
+    }
 
-    Document* doc = owner->GetExtantDoc();
+    Document* doc = window->GetExtantDoc();
     if (nsContentUtils::ShouldResistFingerprinting(doc)) {
       return;
     }
